DiscordChatExporter.Core/Exporting/Filtering/MessageFilter.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using DiscordChatExporter.Core.Discord.Data;
+using DiscordChatExporter.Core.Exporting.Filtering.Parsing;
+using Superpower;
+using Superpower.Model;
+
+namespace DiscordChatExporter.Core.Exporting.Filtering
+{
+    public abstract partial class MessageFilter
+    {
+        public abstract bool Filter(Message message);
+    }
+
+    public partial class MessageFilter
+    {
+        public static MessageFilter CreateFilter(string key, string value)
+        {
+            return key.ToLower() switch
```

Comment:
Just to avoid any locale shenanigans

```suggestion
            return key.ToLowerInvariant() switch
```